sdk/tables/azure-data-tables/azure/data/tables/aio/_table_service_client_async.py

```diff
@@ -86,6 +86,25 @@ def __init__(
         self._client._config.version = kwargs.get('api_version', VERSION)  # pylint: disable=protected-access
         self._loop = loop
 
+    @classmethod
+    def from_connection_string(
+            cls, conn_str, # type: any
+            **kwargs # type: Any
+    ): # type: (...) -> TableServiceClient
+        """Create TableServiceClient from a Connection String.
+
+        :param conn_str:
+            A connection string to an Azure Storage or Cosmos account.
+        :type conn_str: str
+        :returns: A Table service client.
+        :rtype: ~azure.data.tables.TableServiceClient
+        """
+        account_url, secondary, credential = parse_connection_str(
```

Comment:
nit: please feel free to ignore this, but you can try to create a helper method that both sync and async, and TableClient and TableServiceClient coculd call to avoid duplicating this code 4 times